tree/30_generic_methods/file_copy_from_remote_source_recursion.cf

```diff
@@ -71,6 +71,16 @@ bundle agent file_copy_from_remote_source_recursion(source, destination, recursi
         classes      => classes_generic_two("${old_class_prefix}", "${class_prefix}");
 
   methods:
+    # A known issue prevents cfengine to raise classes
+    # when dealing with empty promiser.
+    # Applying this bundle on a directory with a 0 level of recursion,
+    # no promisers will be found and we will need to define classes by ourselves.
+    iteration_2::
+      "error" usebundle => _classes_failure("${old_class_prefix}"),
+        ifvarclass=> "!${class_prefix}_reached";
```

Comment:
it should have been  ifvarclass=> "!${old_class_prefix}_reached";
class_prefix is null :/